### PR TITLE
[YUNIKORN-2759] Replace %w by Errors.join 

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
original discussion: https://issues.apache.org/jira/browse/YUNIKORN-2262

Errors.join can make the code more performant and readable

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2759

### How should this be tested?
make test

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/ugm/manager.go [hunk 1]
```diff
@@ -338,7 +339,7 @@ func (m *Manager) internalProcessConfig(cur configs.QueueConfig, queuePath strin
 				zap.String("queue path", queuePath),
 				zap.Any("limit max resources", limit.MaxResources),
 				zap.Error(err))
-			return fmt.Errorf("problem in using the max resources settings for queuepath: %s. reason: %w", queuePath, err)
+			return errors.Join(fmt.Errorf("problem in using the max resources settings for queuepath: %s. reason: ", queuePath), err)
```
Comment (pbacsko) [inline pkg/scheduler/ugm/manager.go:342]: nit: replace "%s." with "%s,"

Comment (ryankert01) [reply pkg/scheduler/ugm/manager.go]: Done.

**Review by pbacsko — APPROVED** [review]
+1 (pending tests)

**Comment by pbacsko** [comment]
Please fix the "errors" import

**Review by pbacsko — CHANGES_REQUESTED** [review]
Please fix the "errors" import

**Comment by ryankert01** [comment]
@pbacsko Not sure why the ci failed, it runs okay on my forked github repo!
https://github.com/ryankert01/yunikorn-core/actions/runs/10108316582

**Comment by pbacsko** [comment]
@ryankert01 

this is `manager.go`:
```
	"fmt"
	"strings"
	"sync"
	"errors"
```

The `errors` package should be above `fmt`.

**Comment by ryankert01** [comment]
Got it. Just realize I run ci without the pull-request dispatcher, so it won't run go lint.

**Review by pbacsko — APPROVED** [review]
+1

**Pull request closed without merging**